orbit/pkg/useraction/mdm_migration_darwin.go

```diff
@@ -419,3 +409,64 @@ func (m *swiftDialogMDMMigrator) ShowInterval() error {
 func (m *swiftDialogMDMMigrator) SetProps(props MDMMigratorProps) {
 	m.props = props
 }
+
+func (m *swiftDialogMDMMigrator) getMessageAndFlags() (*bytes.Buffer, []string, error) {
+	var tmpl *template.Template
+	var height string
+	vers, err := m.getMacOSMajorVersion()
+	if err != nil {
+		return nil, nil, err
+	}
+	if vers < 14 {
+		height = "440"
+		tmpl = mdmMigrationTemplatePreSonoma
+	} else {
+		height = "669"
+		tmpl = mdmMigrationTemplate
+	}
+
+	var message bytes.Buffer
+	if err := tmpl.Execute(
+		&message,
+		m.props,
+	); err != nil {
+		return nil, nil, fmt.Errorf("execute template: %w", err)
+	}
+
+	flags := []string{
+		// main button
+		"--button1text", "Start",
+		// secondary button
+		"--button2text", "Later",
+		"--height", height,
+	}
+
+	if m.props.OrgInfo.ContactURL != "" {
+		flags = append(flags,
+			// info button
+			"--infobuttontext", "Unsure? Contact IT",
+			"--infobuttonaction", m.props.OrgInfo.ContactURL,
+			"--quitoninfo",
+		)
+	}
+
+	return &message, flags, nil
+}
+
+// TODO: make this a variable for testing
+func (m *swiftDialogMDMMigrator) getMacOSMajorVersion() (int, error) {
+	cmd := exec.Command("sw_vers", "-productVersion")
+	out, err := cmd.Output()
+	if err != nil {
+		return 0, fmt.Errorf("getting macOS version: %w", err)
+	}
+	parts := strings.SplitN(string(out), ".", 2)
+	if len(parts) != 2 {
+		return 0, fmt.Errorf("parsing macOS major version: expected two parts in macOS version string but got %s", out)
+	}
+	major, err := strconv.Atoi(parts[0])
+	if err != nil {
+		return 0, fmt.Errorf("parsing macOS major version: %w", err)
+	}
```

Comment:
I see this a bit risky, do we have any guarantees that there will be two parts, etc? can we do one (or more) of:

1. instead failing on error upstream, default to one of the two modals, worst case scenario they will see the modal in the UI (which is better than crash and no "My Device")
2. can we do this? https://github.com/fleetdm/fleet/blob/de6e0835209a9a3c7fbcaf5de474255fb59c7e12/orbit/pkg/update/runner.go#L338
3. optionally can we be more loose and try to grab only the major version with or without dot?